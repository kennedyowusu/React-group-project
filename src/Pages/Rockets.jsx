import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { fetchRockets } from '../redux/rockets/rocketsSlice';
import Rocket from '../components/Rocket';

const Rockets = () => {
  const dispatch = useDispatch();
  const rocketData = useSelector((state) => state.rockets.rockets);

  useEffect(() => {
    if (rocketData.length === 0) {
      dispatch(fetchRockets());
    }
  }, [dispatch, rocketData]);

  return (
    <div className="rocket-container">
      {rocketData.map((rocket) => (
        <Rocket
          key={rocket.id}
          rocketName={rocket.rocketName}
          description={rocket.description}
          image={rocket.images[0]}
          id={rocket.id}
          reserved={rocket.reserved}
        />
      ))}
    </div>
  );
};

export default Rockets;
