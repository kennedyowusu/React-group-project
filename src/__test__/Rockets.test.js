import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from '../redux/configureStore';
import Rockets from '../components/Rocket';

describe('Test Rockets Component', () => {
  test('rockets component should match with snapshot', () => {
    const rockets = renderer
      .create(

        <Provider store={configureStore}>
          <Rockets />
        </Provider>,

      )
      .toJSON();

    expect(rockets).toMatchSnapshot();
  });
});
