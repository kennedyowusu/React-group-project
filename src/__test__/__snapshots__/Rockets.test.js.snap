// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Rockets Component rockets component should match with snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexWrap": "wrap",
      "justifyContent": "space-around",
    }
  }
>
  <h1 />
  <img
    style={
      Object {
        "height": "200px",
        "objectFit": "cover",
        "width": "200px",
      }
    }
  />
  <p />
  <button
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "green",
        "border": "none",
        "borderRadius": "5px",
        "color": "white",
        "padding": "10px",
      }
    }
    type="button"
  >
    Reserve Rocket
  </button>
</div>
`;
